app/src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -1466,12 +1472,40 @@ public int getSpanSize(int position) {
         }
 
         getRecyclerView().setLayoutManager(layoutManager);
+        calculateAndUpdateSpanCount(grid);
         getRecyclerView().scrollToPosition(position);
         getAdapter().setGridView(grid);
         getRecyclerView().setAdapter(getAdapter());
         getAdapter().notifyDataSetChanged();
     }
 
+    /**
+     * method will calculate the number of spans required for grid item and will update the span accordingly
+     *
+     * @param isGrid
+     */
+    private void calculateAndUpdateSpanCount(boolean isGrid) {
+        getRecyclerView().getViewTreeObserver().addOnGlobalLayoutListener(
+            new ViewTreeObserver.OnGlobalLayoutListener() {
+                @Override
+                public void onGlobalLayout() {
+                    getRecyclerView().getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                    if (isGrid) {
+                        int viewWidth = getRecyclerView().getMeasuredWidth();
+                        int newSpanCount = viewWidth / GRID_ITEM_DEFAULT_WIDTH;
+                        RecyclerView.LayoutManager layoutManager = getRecyclerView().getLayoutManager();
+                        if (layoutManager instanceof GridLayoutManager) {
+                            if (newSpanCount < 1) {
+                                newSpanCount = DEFAULT_FALLBACK_SPAN_COUNT;
+                            }
+                            ((GridLayoutManager) layoutManager).setSpanCount(newSpanCount);
+                            layoutManager.requestLayout();
+                        }
+                    }
+                }
+            });
+    }
+
```

Comment:
@tobiasKaminsky We are using different width and span counts for Grid items. 
Let me know if this also has to be upstreamed.